src/strategyHelpers/canary/podCanaryHelper.ts

```diff
@@ -22,45 +26,32 @@ export async function deployPodCanary(filePaths: string[], kubectl: Kubectl) {
          const name = inputObject.metadata.name
          const kind = inputObject.kind
 
-         if (isDeploymentEntity(kind)) {
+         if (!onlyDeployStable && isDeploymentEntity(kind)) {
             core.debug('Calculating replica count for canary')
             const canaryReplicaCount = calculateReplicaCountForCanary(
                inputObject,
                percentage
             )
             core.debug('Replica count is ' + canaryReplicaCount)
 
-            // Get stable object
-            core.debug('Querying stable object')
+            const newCanaryObject = canaryDeploymentHelper.getNewCanaryResource(
+               inputObject,
+               canaryReplicaCount
+            )
+            newObjectsList.push(newCanaryObject)
+
+            // if there's already a stable object, deploy baseline as well
+            // TODO: get rid of baseline objects entirely
```

Comment:
what do you mean by this TODO?

Reply:
this was pre-our discussion yesterday, I originally thought it didn't make sense to group baseline with canary since generally a stable deployment will already exist but I see the value of it, so removing it. Thanks for catching that!